src/structures/cluster.cpp

```diff
@@ -80,4 +80,18 @@ int32_t Cluster::get_average_inserted_sequence_size() const
     }
     int32_t average_size = std::round(static_cast<double>(sum_sizes) / members.size());
     return average_size;
-}
\ No newline at end of file
+}
+
+std::vector<Junction> Cluster::get_members() const
+{
+    return members;
+}
+
+bool operator<(Cluster const & lhs, Cluster const & rhs)
+{
+    return lhs.get_average_mate1() != rhs.get_average_mate1()
+        ? lhs.get_average_mate1() < rhs.get_average_mate1()
+        : lhs.get_average_mate2() != rhs.get_average_mate2()
+            ? lhs.get_average_mate2() < rhs.get_average_mate2()
+            : lhs.get_average_inserted_sequence_size() < rhs.get_average_inserted_sequence_size();
```

Comment:
Can you use this std:tie here aswell?

Reply:
Yes, I could but decided against it. The `std::tie` creates one tuple each for the lhs and rhs and thus calls all six methods (including the copy operations needed in there). The current comparison first calls two methods and only if their value is equal calls the next two. So we can decide between elegance and speed here..